src/Orleans.Streaming.RabbitMQ/Providers/Streams/RabbitMQ/RabbitMQAdapter.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Orleans.Serialization;
+using Orleans.Streams;
+
+namespace Orleans.Providers.RabbitMQ.Streams.RabbitMQ
+{
+    internal class RabbitMQAdapter<TDataAdapter> : IQueueAdapter
+        where TDataAdapter : IRabbitMQDataAdapter
+    {
+        private TDataAdapter _dataAdapter;
+        private SerializationManager _serializationManager;
+        private ILoggerFactory _loggerFactory;
+        private HashRingBasedStreamQueueMapper _queueMapper;
+        private RabbitMQOptions _options;
+        private string _serviceId;
+        private string _providerName;
+
+        public string Name { get; }
+        public bool IsRewindable => false;
+        public StreamProviderDirection Direction => StreamProviderDirection.ReadWrite;
+
+        public RabbitMQAdapter(TDataAdapter dataAdapter,
+                               SerializationManager serializationManager,
+                               ILoggerFactory loggerFactory,
+                               HashRingBasedStreamQueueMapper queueMapper,
+                               RabbitMQOptions options,
+                               string serviceId,
+                               string providerName)
+        {
+            _dataAdapter = dataAdapter;
+            _serializationManager = serializationManager;
+            _loggerFactory = loggerFactory;
+            _queueMapper = queueMapper;
+            _options = options;
+            _serviceId = serviceId;
+            _providerName = providerName;
+        }
+
+        public IQueueAdapterReceiver CreateReceiver(QueueId queueId)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Sends messages to RabbitMQ.
+        /// </summary>
+        public Task QueueMessageBatchAsync<T>(Guid streamGuid, string streamNamespace, IEnumerable<T> events, StreamSequenceToken token, Dictionary<string, object> requestContext)
+        {
+            var queueId = _queueMapper.GetQueueForStream(streamGuid, streamNamespace);
+            var logger = _loggerFactory.CreateLogger($"{this.GetType().FullName}.{queueId}");
+            var manager = new RabbitMQManager(_options, logger);
```

Comment:
Manager is created from scratch (with all the connection open, exchange/queue setup logics) for every message batch?

Reply:
This is where I was a bit confused on how the Stream provider worked internally. I wasn't sure if it was a one-time setup and then it polled for messages or it was instantiated on every stream. If that's the case, I can likely slim some of this down a bit, I'll just need to figure out a better configuration story so it's a configure once situation. Any advice on how to do that?

Reply:
Ah, I see how it was [done](https://github.com/zitmen/Orleans.Streams.RabbitMqStreamProvider/blob/master/Orleans.Streams.RabbitMqStreamProvider/SiloBuilderExtensions.cs).